### PR TITLE
Export package.json

### LINKED ISSUE #907: Bug: 3.0.0-beta.7's package.json cannot be required in nodejs ^12.7.0
**Reproduction**

Steps to reproduce the behaviour:

1. Install nodejs 12.7.0 and node-fetch@latest
2. Create a script containing `require('node-fetch/package.json')`
3. Run the script
4. 

```
$ node nf.js
internal/modules/cjs/loader.js:491
  throw new ERR_PACKAGE_PATH_NOT_EXPORTED(basePath, mappingKey);
  ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './package.json' is not defined by "exports" in /Users/wheresrhys/Projects/fetch-mock/node_modules/node-fetch/package.json
    at applyExports (internal/modules/cjs/loader.js:491:9)
    at resolveExports (internal/modules/cjs/loader.js:507:23)
    at Function.Module._findPath (internal/modules/cjs/loader.js:635:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:953:27)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/Users/wheresrhys/Projects/fetch-mock/nf.js:1:1)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}

```

**Expected behaviour**

No error (as is the case with node-fetch@2.x or node-fetch@latest with nodejs<12.7.0)

**Your Environment**

| software         | version
| ---------------- | -------
| node-fetch       | 3.0.0-beta.7
| node             | 12.7.0 (also recreated in 12.8.0 and 12.8.3)
| npm              | 6.14.4
| Operating System | MacOS 10.14.6 (Mojave) + iTerm running bash

**Additional context**

TBH this feels like it could be a bug in nodejs (side effect of [removing the experimental-modules flag?](https://github.com/nodejs/node/releases/tag/v12.17.0)), but thought I'd raise here first as you're more likely to know any context relevant to the node-fetch library I may have missed

### PR DESCRIPTION
**Pull request by wheresrhys**
Fixes #907 (see the issue for context)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,10 @@
 	"sideEffects": false,
 	"type": "module",
 	"exports": {
-		"import": "./src/index.js",
-		"require": "./dist/index.cjs",
+    ".": {
+      "import": "./src/index.js",
+      "require": "./dist/index.cjs"
+    },
     "./package.json": "./package.json"
```
Comment (Richienb) [inline package.json:14]: Are we able to remove this then?

Comment (wheresrhys) [reply package.json:14]: I don't understand the question, sorry

Comment (Richienb) [reply package.json:14]: Are we now able to remove `    "./package.json": "./package.json"`?

Comment (wheresrhys) [reply package.json:14]: That's one of the main things I'm adding in this PR. It's essential to achieving the aims of the PR, namely being able to import/require 'node-fetch/package.json' in nodejs 12.7.0+

**Comment by MylesBorins** [comment]
Member of the Node.js modules team here. It is likely a very good idea to land this change. `package.json` needs to be explicitly exported.

**Comment by wheresrhys** [comment]
@jimmywarting Are you able to release a new beta containing this update?

**Comment by xxczaki** [comment]
@wheresrhys See #912

**Comment by wheresrhys** [comment]
Thank you

**Comment by xxczaki** [comment]
@wheresrhys Released [`v3.0.0-beta.8`](https://github.com/node-fetch/node-fetch/releases/tag/v3.0.0-beta.8).

**Pull request merged**